packages/bot-web-ui/src/pages/bot-builder/quick-strategy/form-wrappers/upgraded-qs-v2/strategy-list.tsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { Link } from '@deriv-com/quill-ui';
+import { Text } from '@deriv/components';
+import { TStrategyValue, TTradeType } from './trade-constants';
+import { localize } from '@deriv/translations';
+
+type TStrategyListProps = {
+    option: number;
+    trade_types: string[];
+    trade_types_values: (TTradeType | TTradeType[])[];
+    search_value: string;
+    is_searching: boolean;
+    onSelectStrategy: (strategy: string, trade_type: string) => void;
+};
+
+const StrategyList = ({
+    option,
+    trade_types,
+    trade_types_values,
+    search_value,
+    is_searching,
+    onSelectStrategy,
+}: TStrategyListProps) => {
+    const [is_result_found, setIsResultFound] = React.useState(true);
+    const filterVisibleContent = (strategy: string) => {
+        const is_strategy_match_value = strategy.toLowerCase().includes(search_value.toLowerCase());
+        return (is_searching && is_strategy_match_value) || !is_searching;
+    };
```

Comment:
Can we memoize the value here based on the dependency? This looks like a heave calculation that will run on every rerender.

Reply:
logic has been rewritten, not the actual